src/hyp3_back_projection/back_projection.py

```diff
@@ -0,0 +1,125 @@
+    """
+    lines = [str(dem_path), str(dem_rsc_path)]
+    with open(output_dir / 'params', 'w') as f:
+        f.write('\n'.join(lines))
+
+
+def back_project_single_granule(granule_path: Path, orbit_path: Path, work_dir: Path) -> None:
+    """Back-project a single Sentinel-1 level-0 granule.
+
+    Args:
+        granule_path: Path to the granule to back-project
+        orbit_path: Path to the orbit file for the granule
+    """
+    required_files = ['elevation.dem', 'elevation.dem.rsc', 'params']
+    for file in required_files:
+        if not (work_dir / file).exists():
+            raise FileNotFoundError(f'Missing required file: {file}')
+
+    args = [str(granule_path.with_suffix('')), str(orbit_path)]
+    utils.call_stanford_module('sentinel/sentinel_scene_cpu.py', args, work_dir=work_dir)
+    patterns = ['*hgt*', 'dem*', 'DEM*', 'q*', '*positionburst*']
+    for pattern in patterns:
+        [f.unlink() for f in work_dir.glob(pattern)]
+
+
+def back_project(
+    granules: Iterable[str],
+    earthdata_username: str = None,
+    earthdata_password: str = None,
+    esa_username: str = None,
+    esa_password: str = None,
+    bucket: str = None,
+    bucket_prefix: str = '',
+    work_dir: Optional[Path] = None,
+) -> Path:
+    """Back-project a set of Sentinel-1 level-0 granules.
+
+    Args:
+        granules: List of Sentinel-1 level-0 granules to back-project
+        earthdata_username: Username for NASA's EarthData service
+        earthdata_password: Password for NASA's EarthData service
+        esa_username: Username for ESA's Copernicus Data Space Ecosystem
+        esa_password: Password for ESA's Copernicus Data Space Ecosystem
+        bucket: AWS S3 bucket for uploading the final product(s)
+        bucket_prefix: Add a bucket prefix to the product(s)
+        work_dir: Working directory for processing
+    """
+    utils.set_creds('EARTHDATA', earthdata_username, earthdata_password)
+    utils.set_creds('ESA', esa_username, esa_password)
+    if work_dir is None:
+        work_dir = Path.cwd()
+
+    print('Downloading data...')
+    bboxs = []
+    back_project_args = []
+    for granule in granules:
+        granule_path, granule_bbox = utils.download_raw_granule(granule, work_dir)
+        orbit_path = utils.download_orbit(granule, work_dir)
+        bboxs.append(granule_bbox)
+        back_project_args.append((granule_path, orbit_path))
+
+    full_bbox = unary_union(bboxs).buffer(0.1)
+    dem_path = dem.download_dem_for_back_projection(full_bbox, work_dir)
+    create_param_file(dem_path, dem_path.with_suffix('.dem.rsc'), work_dir)
+
+    for granule_path, orbit_path in back_project_args:
+        back_project_single_granule(granule_path, orbit_path, work_dir=work_dir)
+
+    utils.call_stanford_module('util/merge_slcs.py', work_dir=work_dir)
+
+    if bucket:
+        gslc_path = list(work_dir.glob('S1*.geo'))[0]
+        upload_file_to_s3(gslc_path, bucket, bucket_prefix)
+
+    print('Done!')
```

Comment:
I might make this more explicit, so it's like `Finished back projection for {granule_name}` or something (just since my runs failed so much and I don't remember which Done! is the final Done! )

Reply:
Good idea, I'll change it.